### PR TITLE
Add allow_blank in uniqueness validations

### PR DESCRIPTION
**Pull request by nishant-cyro**
With respect to PR https://github.com/spree/spree/pull/6767

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://nishant-cyro-add-uniqueness-validation.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://nishant-cyro-add-uniqueness-validation.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: core/app/models/spree/stock_item.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ class StockItem < Spree::Base
     has_many :stock_movements, inverse_of: :stock_item
 
     validates :stock_location, :variant, presence: true
-    validates :variant_id, uniqueness: { scope: [:stock_location_id, :deleted_at] }
+    validates :variant_id, uniqueness: { scope: [:stock_location_id, :deleted_at] }, allow_blank: true
```
Comment (JDutil) [inline core/app/models/spree/stock_item.rb]: presence should be required

Comment (nishant-cyro) [inline core/app/models/spree/stock_item.rb]: @JDutil - presence validation is already there in line no. 9.

Comment (JDutil) [inline core/app/models/spree/stock_item.rb]: @nishant-cyro oh yea didn't see variant on that line as well.

**Pull request merged**